integrations/tensorflow/bindings/python/pyiree/tf/support/tf_utils.py

```diff
@@ -0,0 +1,123 @@
+# Lint as: python3
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Test utilities interop with TensorFlow."""
+
+# pylint: disable=not-callable
+# pylint: disable=invalid-name
+# pylint: disable=missing-docstring
+# pylint: disable=protected-access
+
+import os
+import random
+import re
+import tempfile
+
+from absl import flags
+from absl import logging
+import numpy as np
+from pyiree.tf import compiler
+import tensorflow.compat.v2 as tf
+
+FLAGS = flags.FLAGS
+
+
+def set_random_seed(seed=0):
+  """Set random seed for tf, np and random."""
+  tf.random.set_seed(seed)
+  random.seed(seed)
+  np.random.seed(seed)
+
+
+def compile_tf_module(tf_module,
+                      exported_names=(),
+                      target_backends=(),
+                      artifacts_dir=None,
+                      keep_saved_model=False):
+  """Compiles a TensorFlow tf.Module and optionally saves compilation artifacts.
+
+  Args:
+    tf_module: A tf.Module.
+    exported_names: Iterable of dotted function names to consider for
+      compilation.
+    target_backends: Iterable of string backend names to compile for.
+    artifacts_dir: An optional string pointing to where compilation artifacts
+      should be saved.
+    keep_saved_model: An optional boolean controlling whether or not to keep the
```

Comment:
I don't understand why we need keep_saved_model. I didn't understand the description in your commit message.

Reply:
Since our use of saved models here considered an implementation detail, it seemed reasonable to make the default behavior be that the saved model doesn't persist on disk. This is what the behavior was in `tf_test_utils` if `FLAGS.debug_dir` was not provided. Sometimes it's useful to have the saved model as an artifact though, so that's why I added keeping it on disk as an option.

We could make the default be that it's kept on disk with everything else if `artifacts_dir` is provided?

Reply:
I don't think it is very useful to keep the saved model as an artifact, but I'm fine keeping this. I would bias towards always saving it, if we are going to have it around anyway.

Reply:
Sounds good. We could discuss alternatives later, but right now Oleg's benchmarking code depends on this.